command/connect/envoy/bootstrap_config.go

```diff
@@ -476,14 +480,43 @@ func (c *BootstrapConfig) generateListenerConfig(args *BootstrapTplArgs, bindAdd
 		]
 	}`
 
-	if args.StaticClustersJSON != "" {
-		clusterJSON = ",\n" + clusterJSON
+	// Make sure we do not append the same cluster multiple times, as that will
+	// cause envoy startup to fail.
+	selfAdminClusterExists, err := containsSelfAdminCluster(args.StaticClustersJSON)
```

Comment:
I started off by just doing a `strings.Contains(args.StaticClustersJSON, clusterJSON)`, but decided to actually parse out the cluster names to validate it. I think the string comparison would've worked just fine, but parsing out the JSON and checking cluster names felt more semantically correct.